### DISCUSSION TITLE
Copyright Headers

### DISCUSSION THREAD

**Discussion by justinbeaty** [OpenMage/magento-lts#2276] [General]
I know this has been discussed before, but I would like to propose a change in the license headers:

```php
/**
 * OpenMage
 *
 * NOTICE OF LICENSE
 *
 * This source file is subject to the Open Software License (OSL 3.0)
 * that is bundled with this package in the file LICENSE.txt.
 * It is also available through the world-wide-web at this URL:
 * http://opensource.org/licenses/osl-3.0.php
 * If you did not receive a copy of the license and are unable to
 * obtain it through the world-wide-web, please send an email
 * to license@magento.com so we can send you a copy immediately.
 *
 * DISCLAIMER
 *
 * Do not edit or add to this file if you wish to upgrade OpenMage to newer
 * versions in the future. If you wish to customize OpenMage for your
 * needs please refer to https://www.openmage.org for more information.
 *
 * @category    Mage
 * @package     Mage
 * @copyright  Copyright (c) 2006-2020 Magento, Inc. (http://www.magento.com)
 * @copyright  Copyright (c) 2020-2022 OpenMage (https://www.openmage.org)
 * @license    http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
 */
```

There are a few changes here:

1) Changing `Magento` -> `OpenMage` in a few places that are not part of the license or copyright. They are just instances of the project name and I don't see why that cannot change.

2) Adding a second `@copyright` tag. This would only be added to files that have been changed by OpenMage (probably nearly all files) with the correct year for when the first modification was made until 2022.

If people approve, then I can make a PR.

**Comment by Flyingmana** [comment]
Openmage is not a legal entity, therefore can not claim copyright here

**Reply by justinbeaty** [reply]
How about:

```
Copyright (c) 2020-2022 The OpenMage Contributors (https://www.openmage.org)
```

Ref: https://www.linuxfoundation.org/blog/copyright-notices-in-open-source-software-projects/

**Reply by Flyingmana** [reply]
sounds like commonly accepted alternative

**Reply by luigifab** [reply]
yes yes yes!

**Reply by fballiano** [reply]
it should be "2021-2022" since 2020 was still included in the old copyright but, if that's legally possible (that's too difficult of a legal matter to me) that would be nice

**Reply by Flyingmana** [reply]
another point to mention, as soon as we add another `@copyright` line, we cant remove it anymore from this files till copyright expired. So when doing the decision keep in mind, its a non-reversible one.

**Reply by fballiano** [reply]
And that also means that, in 2023, we'll have a release that changes every single file (bye bye doing a diff between releases) just to update the copyright year :-\

**Reply by addison74** [reply]
I totally agree. That block if it will ever be modified should not require other periodic changes.

**Reply by justinbeaty** [reply]
We can overlap years, that's not a problem.

Instead of updating the copyright on all files every year, I will make the script (which will be included in the PR) set the range of years from the earliest modification, to the most recent one. Therefore, when comparing changes only files that would be included in the diff anyways will have a copyright change. This is also more correct, because if a file only had one change from OpenMage in (for example) 2020, then there's no reason to have to put 2020-2023.

**Reply by justinbeaty** [reply]
Even if we could never remove the copyright notice, I still think it's a good idea to implement this.

In my opinion, it shows we're actively maintained. Say a few years from now in 2025, people (and management) feel better seeing more current year than using an e-commerce platform with the last copyright update of 2020.

Furthermore, if someone ever forks OpenMage, we get the credit of having been part of that new codebase's history.

**Comment by fballiano** [comment]
Also, do we really need the "DISCLAIMER" part in every file? we could spare 6 lines removing it and it doesn't seem to be that necessary in 2022

**Reply by justinbeaty** [reply]
We could remove it in a separate PR. I don't have any strong opinion. It may still serve as a determent to modifying core files, but anyone who has modified core files in the past has clearly ignored that warning.

**Reply by luigifab** [reply]
drop it!

**Reply by justinbeaty** [reply]
I'll make two PRs then. One for copyright and one for dropping that disclaimer.

**Comment by justinbeaty** [comment]
First PR: #2295 to add in new copyright tags

**Comment by justinbeaty** [comment]
Second PR: #2297 to change Magento -> OpenMage and remove DISCLAIMER

**Comment by justinbeaty** [comment]
Question: should we also modify `lib/Mage` and `lib/Magento` ?

In general I ignored the `lib` dir because I do not know what plans to be broken out to composer deps.

**Reply by luigifab** [reply]
Yes! There are some commits.

**Reply by justinbeaty** [reply]
Okay, I updated to not ignore `lib` at all. This updated `lib/Magento`, `lib/Mage`, `lib/Varien` and one file in `lib/flex` (which is gone in 20.0 anyway).

The script only changed files where it saw the `@copyright Magento, LLC` line. So it has not changed files in `app/code/core/Zend`, `lib/Zend`, `lib/mcryptcompat`, `lib/phpseclib`, etc. Best to not change those anyway I think.

**Comment by luigifab** [comment]
Does this script must be executed before each release to update copyright years?
Because for all current PR, I think nobody update or add: `* @copyright  Copyright (c) xyz OpenMage (https://www.openmage.org)`.

**Reply by justinbeaty** [reply]
Ideally yes, it is ran before each release. Either on 1.9.4.x then merged to 20.0 (and script run again on 20.0), or on 1.9.4.x and 20.0 independently with the 1.9.4.x commit not cherry-picked.

But, I do not know how much mental overhead this adds to maintainer's workflow. It could be run every so often instead.

**Comment by addison74** [comment]
We did a great job related to this issue.
